src/sst/core/eli/elementbuilder.h

```diff
@@ -88,29 +81,16 @@ class BuilderLibraryDatabase
 
     static Library* getLibrary(const std::string& name)
     {
-        if ( !libraries ) { libraries = new std::map<std::string, Library*>; }
-        auto iter = libraries->find(name);
-        if ( iter == libraries->end() ) {
-            auto* info         = new Library(name);
-            (*libraries)[name] = info;
-            return info;
-        }
-        else {
-            return iter->second;
-        }
+        static Map libraries; // Database
+        auto&      lib = libraries[name];
+        if ( !lib ) lib = new Library(name);
+        return lib;
```

Comment:
This is the Meyers singleton initializing the map on the first call to `getLibrary()`.

It also uses a reference to the map entry and checks if it's `nullptr`, and if it is (such as if it did not exist before), sets it to allocated memory.